### ISSUE TITLE
Enhance Velero PV restore process via Restic

### ISSUE THREAD

**Issue by ippon-orion** [vmware-tanzu/velero#5168]
**Describe the problem/challenge you have**

In case of restic restore, the initial process restore all the PV in same time, and so, it's longer to restore a service fonctional because all PV must be successded before pod restart.

The idea here, is to have a choice to restore PV in ordered manner like `--backup-resource-priorities=persistentvolumes=pv1,pv2,pv3`.

That’s should permit to have the all bandwith dedicated to restore volumes one by one instead of the parallel restored of all volumes. 
And then, when a PV is successful restored, the pod can be restarted in 'readOnly' to recover the service avaibility.

**Describe the solution you'd like**

One option, is maybe to have a variable to SET at restore, maybe somethings like `--restore-resource-priorities=persistentvolumes=pv1,pv2,pv3`

**Environment:**

- Velero version : v1.9.0
- Kubernetes version (use `kubectl version`): 1.21.0

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by Lyndon-Li** [comment]
@ippon-orion “And then, when a PV is successful restored, the pod can be restarted in 'readOnly'” what does this mean? How could we make a pod in 'readOnly'?

**Comment by ippon-orion** [comment]
Nothings, to do in Velero side. 
When the pod is restarted, then, we can access it without waiting for all pods are restarted in case of Statefuleset deployment by example.

**Comment by Lyndon-Li** [comment]
IMO, we don't recommend users to start the business with a partially restored workload, because it is not generally practical. For some environment, it may be fine, but for others, it is not. For example, during the restore, we may restore some pods, inside which, some controllers may run. The controllers may create other resources, start other pods or do anything else, these operations may be conflict with the ongoing Velero restore, as a result, the restore will fail.
As another fact, we usually define RTO (recovery time objective) to a Restore, that time is a static agreement between the workload administrator and the restore service provider. Restore service providers pursue a shorter RTO for a general solutions that work for all cases, they usually don't tune this case by case or env by env.
There are multiple ways for a backup tool to meet a shorter RTO. Velero as a backup tool is also work on this, in future, there will be features to bring better RTO and RPO.

**Comment by ippon-orion** [comment]
I agree in part, regarding potential data corruption and errors upon recovery.

However, many applications know how to handle these cases depending on the number of UP instances.

It is not necessarily a bad practice to make the service available for read only, once the service is partially restored in my opinion.
Of course, writing is blocked/suspended until the whole thing is restored.

At this time Velero, does not allow to restart all instances synchronously at the same time. As a result, the restarts of the pods are delayed in all cases.

The advantage of being able to order the volumes is to use all the bandwidth per volume and to improve the speed of restoration in some cases.

Let's me know if a miss understand something or whatever.

**Comment by Lyndon-Li** [comment]
Even though it is not on Velero side, could you share me a concrete example of how a service make itself available as Readonly? And what can we do with this service if it is Readonly?

**Comment by Lyndon-Li** [comment]
It seems you need to do something in order to block the requests. When and how are you going to do this during Velero restore?

**Comment by ippon-orion** [comment]
Yes ofc, there is some manual actions to do. 
In case of a full restore (drp like), i'll setup my upstream proxy in first, to block non-GET requests, an then i'll perform a Velero restore's job after.
Once restore is completed successful, i'll disable blocking non-GET requests.

Does this answer help you understand my thinking?

**Comment by Lyndon-Li** [comment]
After discussing in the group, we think the requirement is rational for the current case, but not quite generic.
Therefore, we will keep the issue open and see if we have some other requirements could be merged with the current one, then we will find a solution them altogether.
